src/generic_event_type.cpp

```diff
@@ -0,0 +1,374 @@
+    for( bionic_id cbm : cbms_to_add ) {
+        if( !target.has_bionic( cbm ) ) {
+            target.add_bionic( cbm );
+        }
+    }
+    for( bionic_id cbm : cbms_to_remove ) {
+        if( target.has_bionic( cbm ) ) {
+            target.remove_bionic( cbm );
+        }
+    }
+    for( std::string variable : generic_variables_to_set_true ) {
+        g->generic_variable_map[variable] = true;
+    }
+    for( std::string variable : generic_variables_to_set_false ) {
+        g->generic_variable_map[variable] = false;
+    }
+    for( std::pair<time_duration, generic_event_type_id> pair : events_to_queue ) {
+        generic_event_types::queue_generic_event( pair.first, pair.second );
+    }
+    if( damage != 0 ) {
+        if( target_part.is_valid() ) {
+            target.deal_damage( nullptr, target_part, damage_instance( DT_BASH,
+                                damage ) );
+        } else {
+            for( const bodypart_id &bp : target.get_all_body_parts() ) {
+                target.deal_damage( nullptr, bp, damage_instance( DT_BASH, damage ) );
+            }
+        }
+    }
+    if( weather_change != WEATHER_NULL ) {
+        get_weather().weather_override = weather_change;
+    }
+    if( update_weather ) {
+        get_weather().next_weather = true;
+    }
+
+    target.mod_healthy( healthy );
+    target.mod_rad( radiation );
+    wet_character( target, wet );
+    target.mod_pain( pain );
+    sound( sound_message, sound_effect );
+    target.add_msg_if_player( message );
+}
+
+void generic_event_types::load_pair( const JsonObject &jo, const std::string & )
+{
+    g->generic_events_vector.emplace_back( generic_requirement_type_id( jo.get_string( "require" ) ),
+                                           generic_event_type_id( jo.get_string( "event" ) ) );
+}
+
+void generic_event_types::process_generic_pairs()
+{
+    std::vector<std::pair<time_point, generic_event_type_id>>::iterator queued_event =
+                g->queued_generic_events.begin();
+
+    while( queued_event != g->queued_generic_events.end() ) {
+        if( queued_event->first <= calendar::turn ) {
+            queued_event->second->do_event( );
+            queued_event = g->queued_generic_events.erase( queued_event );
+        } else {
+            ++queued_event;
+        }
+    }
+    for( std::pair<generic_requirement_type_id, generic_event_type_id> require_event :
+         g->generic_events_vector ) {
+        if( require_event.first->test( get_player_character().pos(), get_player_character(),
+                                       WEATHER_NULL ) ) {
+            require_event.second->do_event( );
+        }
+    }
+}
+
+void generic_event_types::reset()
+{
+    generic_event_type_factory.reset();
+}
+
+void generic_event_types::finalize_all()
+{
+    generic_event_type_factory.finalize();
+    for( const generic_event_type &get : generic_event_type_factory.get_all() ) {
+        const_cast<generic_event_type &>( get ).finalize();
+    }
```

Comment:
I think this statement and the subsequent loop are just doing the same thing?

Reply:
Honestly not sure, this was in the original generic_factory I copied to make the weather_type factory and I have carried it forward since.